torch_xla_py/xla_dist.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+"""Tool to distribute training on Cloud TPU Pods."""
+from __future__ import division
+from __future__ import print_function
+
+
+class Worker(object):
+
+  def __init__(self, internal_ip, mach_type, zone, hostname=None):
+    self._internal_ip = internal_ip
+    self._mach_type = mach_type
+    self._zone = zone
+    self._hostname = hostname
+
+
+class Cluster(object):
+
+  def __init__(self, workers):
+    self._workers = workers
+
+  def validate(self):
+    """Validates the current cluster configuration.
+
+    Raises:
+      RuntimeError: If the cluster is misconfigured, this validation will
+      raise an error.
+    """
+    raise NotImplementedError()
+
+
+class ClusterResolver(object):
+  """Cluster Resolver for Client VM and Cloud TPU mesh."""
+
+  def __init__(self,
+               tpus,
+               vms=None,
+               zone=None,
+               project=None):
+    """Creates a new ClusterResolver object."""
+
+    assert tpus, "TPU name list must not be empty."
+
+    self._tpus = tpus
+    self._vms = vms
```

Comment:
Do we expect tpu and vms to have the same length in ClusterResolver? If so we might also need validate it at ClusterResolver level?

Reply:
Yes, for now since we're only doing 1:1 mapping. Yeah more like at the `DistributedExecutor` level (that was a random name) which is the class that would actually start the distributed training.